packages/gatsby/src/internal-plugins/query-runner/query-compiler.js

```diff
@@ -91,14 +100,20 @@ class Runner {
   }
 
   async parseEverything() {
-    // FIXME: this should all use gatsby's configuration to determine parsable
-    // files (and how to parse them)
-    let files = glob.sync(`${this.fragmentsDir}/**/*.+(t|j)s?(x)`, {
-      nodir: true,
-    })
-    files = files.concat(
-      glob.sync(`${this.baseDir}/**/*.+(t|j)s?(x)`, { nodir: true })
-    )
+    const filesRegex = `/**/*.+(t|j)s?(x)`
+    let files = [`${this.base}/src`, `${this.base}/.cache/fragments`]
+      .concat(
+        this.additional.map(additional => path.join(this.base, additional))
```

Comment:
This presumes all the same directory (e.g. node_modules/gatsby-theme, or plugins/gatsby-theme)... not sure if/how monorepos would work, so that's a wrinkle to consider.

Reply:
Sounds like an easy fix--thanks!